node/core/backing/src/lib.rs

```diff
@@ -838,7 +838,7 @@ impl util::JobTrait for CandidateBackingJob {
 
 			let (assignment, required_collator) = match assignment {
 				None => return Ok(()), // no need to work.
-				Some((a, r)) => (a, r),
+				Some(r) => r,
```

Comment:
Why does it makes the readability worse? You see that this match returns a tuple by checking the let statement.

Reply:
It's because to get to the original tuple, you need to chase it back two layers instead of one. `let` => `Some(r)` => `assignment`. 

I don't want to block this PR on this, but you did change this in a way that is semantically identical, so I'll ask again: why make the change in the first place?

Reply:
Because it is less code and easier to read. There is no reason to write more text for this. It makes no sense to write out the entire tuple when you can all of this with one character. 

> It's because to get to the original tuple, you need to chase it back two layers instead of one. let => Some(r) => assignment.

No one reading code in this way, you start with the let and directly see that it expects a tuple. You don't start in the tuple.